src/Integration/Binding/BindingController.cs

```diff
@@ -118,15 +118,36 @@ private void OnBind(BindCommandArgs args)
 
         void IBindingWorkflowExecutor.BindProject(BindCommandArgs bindingArgs)
         {
-            //TODO: CM2 - choose the type of binding
-            var writer = this.host.GetService<IConfigurationProvider>();
-            var workflow = new NewBindingWorkflow(this.host, bindingArgs, writer);
-
+            var workflow = CreateBindingWorkflow(bindingArgs);
+            
             IProgressEvents progressEvents = workflow.Run();
             Debug.Assert(progressEvents != null, "BindingWorkflow.Run returned null");
             this.SetBindingInProgress(progressEvents, bindingArgs);
         }
 
+        internal /* for testing purposes */ IBindingWorkflow CreateBindingWorkflow(BindCommandArgs bindingArgs)
+        {
+            //Choose the type of binding
+            var configProvider = this.host.GetService<IConfigurationProvider>();
+            configProvider.AssertLocalServiceIsNotNull();
+
+            var currentConfiguration = configProvider.GetConfiguration();
+            if (currentConfiguration.Mode == SonarLintMode.LegacyConnected)
+            {
+                host.Logger.WriteLine(Strings.Bind_UpdatingLegacyBinding);
+                return new BindingWorkflow(host, bindingArgs);
+            }
+
+            // If we are currently in standalone then the project is being bound for the first time.
+            // If we are in connected mode then the binding is being updated.
```

Comment:
If we are in legacy connected mode? Because I don't think we should do that as we are not cleaning up.

Reply:
I don't understand the question. Legacy mode is handled in line 135; it's called when updating a solution that is currently in legacy bound mode (either because the user has selected _update_ in the Team Explorer, or from the gold bar in the error list controller.

Reply:
I thought we were actually switching mode.